src/libs/E2E/tests/linkingTest.e2e.ts

```diff
@@ -23,6 +24,29 @@ const test = (config: NativeConfig) => {
             return waitForAppLoaded().then(() => E2EClient.submitTestDone());
         }
 
+        const [appearMessagePromise, appearMessageResolve] = getPromiseWithResolve();
+        const [switchReportPromise, switchReportResolve] = getPromiseWithResolve();
+
+        Promise.all([appearMessagePromise, switchReportPromise])
+            .then(() => {
+                console.debug('[E2E] Test completed successfully, exiting…');
+                E2EClient.submitTestDone();
+            })
+            .catch((err) => {
+                console.debug('[E2E] Error while submitting test results:', err);
+            });
+
+        const subscription = DeviceEventEmitter.addListener('onViewableItemsChanged', (res) => {
```

Comment:
Would it fire when we open the first chat? It looks like it would, so we can skip it or change the log '[E2E] Message verification failed,' as it may look like an error, but it would be expected behavior since we only want to check it for entry.name === SWITCH_REPORT